components/sup/src/config.rs

```diff
@@ -127,7 +123,7 @@ impl GossipListenAddr {
 impl Default for GossipListenAddr {
     fn default() -> GossipListenAddr {
         GossipListenAddr(socket_addr_env_or_default(
-            DEFAULT_ADDRESS_ENVVAR,
+            GOSSIP_LISTEN_ADDRESS_ENVVAR,
```

Comment:
Even though this particular `Default` implementation is a bit odd by relying on environment variables (which I suspect we'll want to change in a future refactoring, in order to more clearly delineate where in the code that environment variables' effects can be felt), it would be good to take advantage of the string default that you have created and harmonize that with this. That way we don't have two potentially divergent defaults in the codebase.

(It may be worth investigating transitioning this to `EnvConfig` to see if that helps.)